### PR TITLE
fiche: init at version 0.9.1

### PR DESCRIPTION
**Pull request by pinpox**
###### Motivation for this change
Include [fiche](https://github.com/solusipse/fiche), a command line pastebin service for sharing terminal output with minimal dependencies.

**Note**: this is my first PR to nixpkgs, feedback is welcome so please let me know what has to be changed!

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/fiche/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  installPhase = ''
+    mkdir -p $out/bin
+    mv fiche $out/bin
+    '';
+
+  doCheck = true;
+
+  meta = with stdenv.lib; {
+    description = "Command line pastebin for sharing terminal output.";
```
Comment (SuperSandro2000) [inline pkgs/servers/fiche/default.nix:20]:
```suggestion
    description = "Command line pastebin for sharing terminal output";
```

File: pkgs/servers/fiche/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  version = "0.9.1";
+
+  src = fetchurl {
+    url = "https://github.com/solusipse/${pname}/archive/${version}.tar.gz";
+    sha256 = "TODO";
+  };
+
+  installPhase = ''
+    mkdir -p $out/bin
+    mv fiche $out/bin
```
Comment (SuperSandro2000) [inline pkgs/servers/fiche/default.nix:14]: You can use install here to also set the permission correct.

File: pkgs/servers/fiche/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
+{ stdenv, fetchurl }:
+
+stdenv.mkDerivation rec {
+  pname = "fiche";
+  version = "0.9.1";
+
+  src = fetchurl {
+    url = "https://github.com/solusipse/${pname}/archive/${version}.tar.gz";
+    sha256 = "TODO";
```
Comment (SuperSandro2000) [inline pkgs/servers/fiche/default.nix:9]: If you try to built this locally you will get the Sha that needs to be put here.

File: pkgs/servers/fiche/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
+{ stdenv, fetchurl }:
+
+stdenv.mkDerivation rec {
+  pname = "fiche";
+  version = "0.9.1";
+
+  src = fetchurl {
```
Comment (SuperSandro2000) [inline pkgs/servers/fiche/default.nix:7]: Please use ``fetchFromGitHub``.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/servers/fiche/default.nix (hunk 4 above)
Comment (pinpox) [reply pkgs/servers/fiche/default.nix]: Updated.

File: pkgs/servers/fiche/default.nix (hunk 3 above)
Comment (pinpox) [reply pkgs/servers/fiche/default.nix]: Done.

File: pkgs/servers/fiche/default.nix (hunk 2 above)
Comment (pinpox) [reply pkgs/servers/fiche/default.nix:16]: Could you elaborate? I don't understand what I have to change.

**Comment by pinpox** [comment]
The package builds and works fine now on my machine.

File: pkgs/servers/fiche/default.nix (hunk 2 above)
Comment (turboMaCk) [reply pkgs/servers/fiche/default.nix:16]: see https://github.com/solusipse/fiche/blob/master/Makefile#L9

**Comment by pinpox** [comment]
I'd like to add a service aswell, so that this can be used declaratively from your `configuration.nix` like:
```nix
services = {
  fiche = {
    enable = true;
    domain = "example.com";
    listen_addr  = "0.0.0.0";
    port = "9999";
    slug size = 3;
    output directory = "/var/dir/";
    buffer size = 2048;
    user name = fiche;
    log file = home/www/fiche-log.txt;
    banlist = [];
    whitelist = [];
    https = true; # -S command line option
  };
};
```

Should that be a separate PR? and how/where do I add such a configuration option?

**Comment by turboMaCk** [comment]
I would recommend doing separate PR for service. It will likely require other set of reviewers. I would also recommend trying to first define service as part of your personal configuration first and only then trying to port it to nixos modules.

File: pkgs/servers/fiche/default.nix (hunk 2 above)
Comment (pinpox) [reply pkgs/servers/fiche/default.nix:16]: So just replace those two lines with this?
```
install -m 0755 fiche $out
```

Comment (pinpox) [reply pkgs/servers/fiche/default.nix]: I have pushed a commit with the suggested changes

File: pkgs/servers/fiche/default.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+  src = fetchFromGitHub {
+    owner = "solusipse";
+    repo = "fiche";
+    rev = version;
+    sha256 = "1102r39xw17kip7mjp987jy8na333gw9vxv31f7v8q05cr7d7kfb";
+  };
+
+  installPhase = ''
+    install -m 0755 fiche $out/bin
```
Comment (turboMaCk) [inline pkgs/servers/fiche/default.nix:16]:
```suggestion
    mkdir -p $out/bin
    install -m 0755 fiche $out/bin
```

to avoid:

```
install: cannot create regular file '/nix/store/z026bmn69l1lbjgv09vgikqq67b8bnwr-fiche-0.9.1/bin': No such file or directory
```

**Comment by turboMaCk** [comment]
@pinpox Please squash commits into a single one with the message `fiche: init at version 0.9.1` and I think this is good to go from my point of view.

**Comment by pinpox** [comment]
> @pinpox Please squash commits into a single one with the message `fiche: init at version 0.9.1` and I think this is good to go from my point of view.

I found a small bug, fixed now and squashed.

**Comment by turboMaCk** [comment]
yup that's cool. I also found it but looks I didin't have to point it out https://github.com/NixOS/nixpkgs/pull/101469#discussion_r510972076

:+1: good work on this. It will probably take a few days for committers to merge this because they're quite overloaded with PRs so please be patient.

File: pkgs/servers/fiche/default.nix [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  src = fetchFromGitHub {
+    owner = "solusipse";
+    repo = "fiche";
+    rev = version;
+    sha256 = "1102r39xw17kip7mjp987jy8na333gw9vxv31f7v8q05cr7d7kfb";
+  };
+
+  installPhase = ''
+    mkdir -p $out/bin
+    install -m 0755 fiche $out/bin
```
Comment (sikmir) [inline pkgs/servers/fiche/default.nix:16]:
```suggestion
    install -m 0755 fiche -t $out/bin
```

**Comment by pinpox** [comment]
Can someone merge this or comment on what is missing?

File: maintainers/maintainer-list.nix [hunk 7]
```diff
@@ -6751,6 +6751,16 @@
       fingerprint = "A3A3 65AE 16ED A7A0 C29C  88F1 9712 452E 8BE3 372E";
     }];
   };
+  pinpox = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:6754]: Please put this into a separate commit with the message ``maintainers: add pinpox``.

File: pkgs/top-level/all-packages.nix [hunk 8]
```diff
@@ -142,6 +142,8 @@ in
 
   ankisyncd = callPackage ../servers/ankisyncd { };
 
+  fiche = callPackage ../servers/fiche { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:145]: It would be great if you could sort this into a fitting category and alphabetically there.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Also please squash the commits related to the package together into the init commit.

otherwise LGTM

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101469` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>fiche</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101469` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>fiche</li>
  </ul>
</details>
